src/core/telemetry/utils.ts

```diff
@@ -5,13 +5,31 @@ import TelemetryReporter from "./telemetryReporter";
 import type { TelemetryEventMeasurements, TelemetryEventProperties } from "./telemetryReporterTypes";
 import { readCLIEnvFile } from "../utils";
 import * as crypto from "crypto";
+import os from "os";
+import { getMachineIdForTelemetry } from "../swa-cli-persistence-plugin/impl/machine-identifier";
+import { DEFAULT_CONFIG } from "../../config";
 
 const aiKey = "8428a7f6-6650-4490-a15a-c7f7a16449d7";
+const pkg = require("../../../package.json");
 
 export async function collectTelemetryEvent(event: string, properties?: TelemetryEventProperties, measurements?: TelemetryEventMeasurements) {
   const reporter = await getTelemetryReporter();
+
   if (reporter) {
-    reporter.sendTelemetryEvent(event, properties, measurements);
+    let macAddressHash = (await getMachineIdForTelemetry()).toString();
+    let extendedTelemetryEventProperties = {
+      macAddressHash: macAddressHash,
+      installationId: crypto
+        .createHash("sha256")
+        .update(pkg.version + macAddressHash)
+        .digest("hex"),
+      subscriptionId: DEFAULT_CONFIG.subscriptionId!,
+      sessionId: getSessionId(new Date().getTime()),
```

Comment:
Q: Should we have same sessionId if two telemetry ponts are captured in the same session.. This will not have same session. ?

Reply:
but in any possible case flow, we'll only be collecting one telemetry point right?

Reply:
It might be possible to collect one or more telemetry point for a flow. e.g. we starte perf measures and then these logs or multiple log points for the same flow.. in that case the sessionId will be different.

 So in that case we should have same sessionId for both telemetry points..

Reply:
yes my bad, got your point, will update